titan/features/high_risk.py

```diff
@@ -81,59 +69,40 @@ def update_agent(self, model: "model.HIVModel"):
             model: the instance of HIVModel currently being run
         """
         if not self.active:
-            return None
-
-        if self.duration > 0:
+            # released last step
+            if self.agent.incar.release_time == model.time - 1:  # type: ignore[attr-defined]
+                self.become_high_risk(model.pop, model.time)
+
+            # incarcerated last step
```

Comment:
This could probably use explanatory comments

Reply:
updated - better?